vessim/sil.py

```diff
@@ -143,28 +118,27 @@ def start(self, microgrid: Microgrid) -> None:
 
     def step(self, time: datetime, p_delta: float, actor_infos: dict) -> None:
         assert self.microgrid is not None
-        self.broker._add_microgrid_data(
-            time,
-            {
-                "microgrid": self.microgrid,
-                "actor_infos": actor_infos,
-                "p_delta": p_delta,
-            },
+        self.data_pipe_in.send(
+            (
+                time,
+                {
+                    "microgrid": self.microgrid,
+                    "actor_infos": actor_infos,
+                    "p_delta": p_delta,
+                },
+            )
         )
 
-    def finalize(self) -> None:
-        self.broker._finalize()
-
     def _collect_set_requests_loop(self):
         while True:
             events_by_category = defaultdict(dict)
-            for event in self.broker._consume_events():
+            while self.events_pipe_out.poll():
+                event = self.events_pipe_out.recv()
                 events_by_category[event["category"]][event["time"]] = event["value"]
-            for category, events in events_by_category.items():
+            for category, _ in events_by_category.items():
                 self.request_collectors[category](
                     events=events_by_category[category],
```

Comment:
Looking at this a second time, this would be nicer as either:
```suggestion
            for category in events_by_category.keys():
                self.request_collectors[category](
                    events=events_by_category[category],
```
or as:
```suggestion
            for category, events in events_by_category.items():
                self.request_collectors[category](
                    events=events,
```